aspnetcore/signalr/hubs/sample/wwwroot/js/chat.js

```diff
@@ -0,0 +1,52 @@
+﻿const connection = new signalR.HubConnection('/chathub');
+
+connection.on('ReceiveMessage', (timestamp, user, message) => {
+    const encodedUser = user;
+    const encodedMsg = message;
+    const listItem = document.createElement('li');
+    listItem.innerHTML = timestamp + ' <b>' + encodedUser + '</b>: ' + encodedMsg;
+    document.getElementById('messages').appendChild(listItem);
+});
+
+document.getElementById('send').addEventListener('click', event => {
+    const msg = document.getElementById('message').value;
+    const usr = document.getElementById('user').value;
+    const grp = document.getElementById('group').value;
+    if (grp != null) {
+        connection.invoke('SendMessageToGroup', usr, msg, grp).catch(err => showErr(err));
```

Comment:
use `send`, `invoke` is for when you expect a response from the server

Reply:
But then `catch` doesn't work. The point here is to illustrate the standard flow, and I think `.invoke` is the standard flow. `.send` is a more advanced scenario IMO.